17.md

```diff
@@ -0,0 +1,148 @@
+    ...
+  ],
+  "content": "<message-in-plain-text>",
+}
+```
+
+`.content` MUST be plain text and tags that mention, quote and assemble threading structures MUST follow [NIP-10](10.md). Fields `id` and `created_at` are required. 
+
+The set of `pubkey` + `p` tags defines a group chat. Clients SHOULD render messages of the same group in a continuous thread. If a new `p` tag is added or the current one is removed, a new group is created. 
+
+An optional `subject` tag defines the current name/topic of the conversation. Any member can change the topic by simply submitting a new `subject` to an existing `pubkey` + `p`-tags group. There is no need to send `subject` in every message. The newest `subject` in the thread is the subject of the conversation. 
+
+Kind `14`s MUST never be signed. If it is signed, the message might leak to relays and become **fully public**.
+
+## Private Message Protocol
+
+Following [NIP-59](59.md), the **unsigned** `kind:14` chat message must be sealed (`kind:13`) and then gift wrapped (`kind:1059`) to each receiver and the sender individually.
+
+```js
+{
+  "id": "<usual hash>",
+  "pubkey": randomPublicKey,
+  "created_at": randomTimeUpTo2DaysInThePast(), 
+  "kind": 1059, // gift wrap
+  "tags": [
+    ["p", receiverPublicKey, "<relay-url>"] // receiver
+  ],
+  "content": nip44Encrypt(
+    {
+      "id": "<usual hash>",
+      "pubkey": senderPublicKey,
+      "created_at": randomTimeUpTo2DaysInThePast(), 
+      "kind": 13, // seal
+      "tags": [], // no tags
+      "content": nip44Encrypt(unsignedKind14, senderPrivateKey, receiverPublicKey),
+      "sig": "<signed by senderPrivateKey>"
+    },
+    randomPrivateKey, receiverPublicKey
+  ),
+  "sig": "<signed by randomPrivateKey>"
+}
+```
+
+The encryption algorithm MUST use the latest version of [NIP-44](44.md). 
+
+Clients MUST verify if pubkey of the `kind:13` is the same pubkey on the `kind:14`, otherwise any sender can impersonate others by simply changing the pubkey on `kind:14`.
+
+Clients SHOULD randomize `created_at` in up to two days in the past in both the seal and the wrap to make sure grouping by `created_at` doesn't reveal the group's metadata. 
+
+The gift wrap's `p`-tag can be the receiver's main pubkey or an alias key created to receive DMs without exposing the receiver's identity.
+
+Clients CAN offer disappearing messages to their own users by not creating a GiftWrap to the sender's public key or by setting an `expiration` tag in the gift wrap of each receiver.   
+
+## Publishing 
+
+Kind `10050` indicates the user's preferred relays to receive DMs. The event MUST include a list of `relay` tags with relay URIs.
+
+```js
+{
+  "kind": 10050,
+  "tags": [
+    ["relay", "wss://inbox.nostr.wine"],
+    ["relay", "wss://myrelay.nostr1.com"],
+  ],
+  "content": "",
+  //...other fields
+}
+```
+
+Clients SHOULD publish kind `14` events to the `10050`-listed relays, falling back to `read` relays of [NIP-65](65.md) if `kind:10050` is not available.
```

Comment:
```suggestion
Clients MUST publish kind `14` events to the `10050`-listed relays, refusing to send them if that is not available.
```

Reply:
I am torn on this. It forces DM clients to only work with each other. 

But it also forces the creation of 10050 events by every DM client. It forces the issue that these clients MUST help the use manage a 10050, instead of just half implementing it and using NIP-65 lists or kind3.

Reply:
I don't comprehend the reason for specific per-NIP relay lists. Why doesn't a general "this relay is where the user accepts private events" work? Do different DM specs really have different requirements for the relays? I haven't heard of any. Are we just extending a general principle of complete separateness so as to avoid compatibility issues?

Reply:
@mikedilger 2 separate concerns: 

1. The more I think about this, the more I see this need to define relay addresses per NIP. There are things like Public Chats (NIP-26), for instance where NIP-65 should be ignored and the relay to be used is the one what was setup in the head-event's metadata. A similar thing happens for @staab's private communities. Clients should follow the community relay, not the user's personal relay. Medical Data, for instance, might have a relay marker to make sure the relay is HIPAA compliant (US-only). So, more private stuff might need additional relay information. This makes relay categories in NIP-65 not only difficult to predict but it might grow the list significantly. 

2. On the other way, we do have relays today that can be used to receive any private event, like `inbox.nostr.wine`. That shouldn't be just for the DMs in this particular PR.

Reply:
All those things you mention in (1) don't conflict. They can all have their own way of choosing relays. It sounds like they already do. NIP-65 is only for when a case can be generalized. And it seems to me it would still be useful to have a generalized "private event" relay, even if none of the cases you list can utilize it.

If this NIP-17 can't utilize it either, I'm not understanding why not. If it turns out there is no general case, that all private event related NIPs need their own special thing, then you win and I lose and I'll bow out. It just seems to me that I will always be choosing the same relays for all these cases, relays that honor not sending private events to anybody but me, and allowing me to delete them.

Reply:
Actually, I agree with you, @mikedilger, that we can generalize the DM version to the private **event** from your NIP-65 change. It fits. 

Over there, I would change the description from `Direct Message` to simply `Private Inbox` and use the letter `p`. Clients should expect that Private Inbox relays accept any incoming event that tags one of their protected users. Only that user can download the event after AUTH. 

@staab @fiatjaf what do you say? Do we really need something specific just for DMs?

Reply:
I was writing a big argument on why specific is better than generic, but I deleted it. I think it's just think it's a good rule of thumb because specific always be made generic later, but not the opposite. What if we come up with a new use case that doesn't fit in the "generic private relay" category?

I don't know. The argument is not very strong. Whatever you decide is fine by me.

Reply:
I agree with fiatjaf, we might as well start with this NIP and expand it if needed. I do think the way Vitor articulated it, "Clients should expect that Private Inbox relays accept any incoming event that tags one of their protected users." shouldn't cause any problems. I also like including the recommendation that only recipients be able to download the events. But it could go in either place.

Reply:
Ok, let's keep this as is and after @mikedilger's PR gets merged, we can change it to revert to the Private Inbox relays in NIP-65 instead of the read relays.